core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.kafka
+
+import java.util
+import java.util.regex.Pattern
+
+import akka.Done
+import akka.actor.{Actor, ActorRef, Cancellable, Props, Status}
+import akka.event.LoggingReceive
+import org.apache.kafka.clients.consumer._
+import org.apache.kafka.common.TopicPartition
+
+import scala.collection.JavaConverters._
+import scala.concurrent.Promise
+
+object KafkaConsumerActor {
+  case class StoppingException() extends Exception("Kafka consumer is stopping")
+  def props[K, V](settings: ConsumerSettings[K, V]) = Props(new KafkaConsumerActor(settings))
```

Comment:
explicit return type in all public api, so that it's not accidentally changed
`: Props`